File: frontend/src/app/pages/services/customer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient} from '@angular/common/http';
import { Urls } from './url';
import {Observable} from "rxjs";
@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  API_URL = new Urls();
  constructor(private http: HttpClient) { }


  gtCustomers ():Observable<any> {
    return this.http.get(`${this.API_URL.url}/CustomerService/viewCustomers`);
  }
  getCustomersToApprove ():Observable<any> {
    return this.http.get(`${this.API_URL.url}/CustomerService/viewCustomersToApprove`);
  }
  getCustomersToApproveDelete ():Observable<any> {
    return this.http.get(`${this.API_URL.url}/CustomerService/viewCustomersToApproveDelete`);
  }
  addCustomer(Customer):Observable<any>  {
    return  this.http.post(`${this.API_URL.url}/CustomerService/addCustomers`, Customer);
  }
  updateCustomer(Customer):Observable<any> {
    return this.http.put(`${this.API_URL.url}/CustomerService/updateCustomer` ,Customer);
  }
  approveCustomer(Customer):Observable<any> {
    return this.http.put(`${this.API_URL.url}/CustomerService/approveCustomer` ,Customer);
  }
  deleteCustomer(Customer):Observable<any> {
    return this.http.put(`${this.API_URL.url}/CustomerService/deleteCustomer` ,Customer);
  }
  approveDeleteCustomer(Customer):Observable<any> {
    return this.http.put(`${this.API_URL.url}/CustomerService/approveDeleteCustomer` ,Customer);
  }



}

File: backend/src/main/java/com/revenue/revenueCollection/WebService/CustomerService.java
package com.revenue.revenueCollection.WebService;

import com.revenue.revenueCollection.Interfaces.CustomerBal;

import com.revenue.revenueCollection.Models.*;
import com.revenue.revenueCollection.Models.Customer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import javax.ejb.EJB;
import javax.validation.Valid;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.util.List;

@RestController
@RequestMapping(value="/api/CustomerService/")
public class CustomerService {
    @Autowired
    private CustomerBal customerBal;






    @GetMapping(path="/viewCustomers",produces = "application/json")
    public List<Customer> getAllCustomers() {
        return customerBal.viewCustomerDetails();
    }


    @GetMapping(path="/viewCustomersToApprove",produces = "application/json")
    public List<Customer> getCustomerToapprove() {
        return customerBal.getCustomerToApprove();
        //return Response.status(200).entity(apps).build();

    }
    @GetMapping(path="/viewCustomersToApproveDelete",produces = "application/json")
    public List<Customer> getCustomersToApproveDelete() {
        return customerBal.getCustomerToApproveDelete();
        //return Response.status(200).entity(apps).build();

    }
    
    @PostMapping("/addCustomers")
    public Response addCustomer(@Valid @RequestBody Customer customer){

        ResponseMessage response = customerBal.registerCustomer(customer);
        return Response.status(200).entity(response).build();
    }

    @PutMapping(value = "/updateCustomer",consumes = "application/json",produces = "application/json")
    public Response updateCustomerDetails(
                                       @Valid @RequestBody Customer customer) {

        ResponseMessage response = customerBal.updateCustomer(customer);
        return Response.status(200).entity(response).build();
    }



    @PutMapping(path="/approveCustomer" ,consumes = "application/json",produces = "application/json")
    public Response approveCustomer(@Valid @RequestBody Customer customers) {
       
        ResponseMessage response = customerBal.approveCustomer(customers);

        return Response.status(200).entity(response).build();
    }
    @PutMapping(path="/deleteCustomer" ,consumes = "application/json",produces = "application/json")
    public Response deleteCustomer(@Valid @RequestBody Customer Customers) {
        ResponseMessage response = customerBal.deleteCustomer(Customers);
        return Response.status(200).entity(response).build();
    }
    @PutMapping(path="/approveDeleteCustomer" ,consumes = "application/json",produces = "application/json")
    public Response approveDeleteCustomer(@Valid @RequestBody Customer Customers) {
        ResponseMessage response = customerBal.approveDeleteCustomer(Customers);
        return Response.status(200).entity(response).build();
    }
}
